src/smt/set_defaults.cpp

```diff
@@ -684,10 +684,9 @@ void SetDefaults::setDefaultsPost(const LogicInfo& logic, Options& opts) const
     }
   }
 
-  // until bugs 371,431 are fixed
-  if (!opts.prop.minisatUseElimWasSetByUser)
+  if (!opts.prop.minisatVarElimWasSetByUser)
```

Comment:
Don't we also have to guard for `minisatSimp` here?

Reply:
No. The "varElim" option was added (it does not exist on original Minisat) specifically to allow all other simplifications except variable elimination, which is problematic for models and the other stuff in this block.

Reply:
The description of `minisatSimp` says `use Minisat variable and clause elimination, plus other simplifications` though?

Reply:
Yes. So the behavior is that if `minisatSimp` *and* `minisatVarElim` are on, the solver will do variable and clause elimination and some other stuff. If `minisatSimp` is off, does nothing. If `minisatSimp` is on but `minisatVarElim` is off, does all the simplifications except for variable elimination.

Changing the helper of `minisatVarElim` to `"use Minisat variable elimination (requires --minisat-simplification)" would help? It'd not account for the disabling of var elim even when `minisatSimp` is on, though, but I think it'd get convoluted to explain this is the helper.

Reply:
So I think this should be made more clear in both help messages. Right now it is implying a very different behavior (them being independent, and both enabling variable elimination).

Reply:
I gave it a shot, PTAL